src/v/cluster/rm_stm.h

```diff
@@ -221,13 +221,25 @@ class rm_stm final : public persisted_stm<> {
       model::producer_identity, model::tx_seq, model::timeout_clock::duration);
     ss::future<tx_errc> abort_tx(
       model::producer_identity, model::tx_seq, model::timeout_clock::duration);
-
+    /**
+     * Returns the next after the last one decided. If there are no ongoing
+     * transactions this will return next offset to be applied to the the stm.
+     */
     model::offset last_stable_offset();
     ss::future<fragmented_vector<rm_stm::tx_range>>
       aborted_transactions(model::offset, model::offset);
 
     model::offset max_collectible_offset() override {
-        return last_stable_offset();
+        const auto lso = last_stable_offset();
```

Comment:
that should probably be called `first_unstable_offset`

Reply:
not in this PR of cause

Reply:
i agree, it is called `lso` in kafka, hence the name